From 3359637e11ae4eb47e445334e4054efd00857553 Mon Sep 17 00:00:00 2001
From: Oleg Karfich <oleg.karfich@wago.com>
Date: Mon, 14 Aug 2017 12:30:47 +0200
Subject: [PATCH] usb: serial: vendor mismatch; fix compile warning

	due to kernel version 4.9.y the commit
	21586e4dcc9b815555a848f6151e96247f1c2f3f add
	quectel vendor id and seperates quectel devices that
	use qualcomm vendor id.
---
 drivers/usb/serial/option.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index f546eff..211f55e 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -78,11 +78,6 @@ static void option_instat_callback(struct urb *urb);
 #define OPTION_PRODUCT_ETNA_KOI_MODEM		0x7100
 #define OPTION_PRODUCT_GTM380_MODEM		0x7201
 
-/* added by wago */
-#define QUECTEL_VENDOR_ID			0x05c6
-#define QUECTEL_PRODUCT_UC20			0x9003
-#define QUECTEL_PRODUCT_EC20			0x9215
-
 #define HUAWEI_VENDOR_ID			0x12D1
 #define HUAWEI_PRODUCT_E173			0x140C
 #define HUAWEI_PRODUCT_E1750			0x1406
@@ -247,6 +242,7 @@ static void option_instat_callback(struct urb *urb);
 #define UBLOX_PRODUCT_R410M			0x90b2
 /* These Yuga products use Qualcomm's vendor ID */
 #define YUGA_PRODUCT_CLM920_NC5			0x9625
+#define QUECTEL_PRODUCT_EC20			0x9215
 
 #define QUECTEL_VENDOR_ID			0x2c7c
 /* These Quectel products use Quectel's vendor ID */
@@ -571,8 +567,6 @@ static void option_instat_callback(struct urb *urb);
 
 
 static const struct usb_device_id option_ids[] = {
-	{ USB_DEVICE(QUECTEL_VENDOR_ID, QUECTEL_PRODUCT_UC20) },
-	{ USB_DEVICE(QUECTEL_VENDOR_ID, QUECTEL_PRODUCT_EC20) },
         { USB_DEVICE(0x2c7c, 0x0125) }, /* Quectel EC25, EC20 R2.0  Mini PCIe */
         { USB_DEVICE(0x2c7c, 0x0121) }, /* Quectel EC21 Mini PCIe */
 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_COLT) },
@@ -1078,6 +1072,7 @@ static const struct usb_device_id option_ids[] = {
 	{ USB_DEVICE(QUALCOMM_VENDOR_ID, 0x9000)}, /* SIMCom SIM5218 */
 	/* Quectel products using Qualcomm vendor ID */
 	{ USB_DEVICE(QUALCOMM_VENDOR_ID, QUECTEL_PRODUCT_UC15)},
+	{ USB_DEVICE(QUALCOMM_VENDOR_ID, QUECTEL_PRODUCT_EC20)},
 	{ USB_DEVICE(QUALCOMM_VENDOR_ID, QUECTEL_PRODUCT_UC20),
 	  .driver_info = RSVD(4) },
 	/* Yuga products use Qualcomm vendor ID */
-- 
2.7.4

